config/creatures/necropolis.json

```diff
@@ -92,8 +92,8 @@
 		{
 			"regenerate" : 
 			{
-				"type" : "FULL_HP_REGENERATION",
-				"subtype" : 1
+				"type" : "HP_REGENERATION",
+				"val" : 1000
```

Comment:
Don't like arbitrary constants in code.
I get what you're doing here, but still looks weird.

Personally I would've added subtype 1 which provides percentage regeneration and let Wights have 100% regeneration.
Clear intend and additional functionality that modders can use

Reply:
@IvanSavenko maybe do like thelazy suggests? https://heroes.thelazy.net/index.php/Regeneration? They say than in OH3 there is only 50HP regen, so, set it for both trolls and wights?

Reply:
Additional functionality - maybe postpone?

Reply:
50 HP - OK I guess?

Reply:
@IvanSavenko I think it should match OH3 it this behaviour. So, done this. Actually, according to TheLazy, there was a deviation from OH3 behaviour, because trolls with Elixir of Live will have more than 50 HP, and hence be able to regenerate more. So, fix it as 50 HP.